core/store/src/trie/mem/iter.rs

```diff
@@ -0,0 +1,261 @@
+use super::node::{MemTrieNodePtr, MemTrieNodeView};
```

Comment:
Is this file mostly a copy of the iterator implementation of disk trie? What is different here?

Reply:
Yeah it's basically the same except mapping the structures over to memtrie versions. Most notable differences are (1) it doesn't support the extra options like remembering nodes or pruning; (2) it uses None to represent an "empty" node rather than TrieNode::Empty, and in memtrie the Branch and BranchWithValue are separate enum cases.; (3) memtrie code paths don't return any errors.

Reply:
@robin-near could you add what you commented to the beginning of this file so that people who read it in the future won't ask the same question again :)

Reply:
Done.